### PR TITLE
[OpenVINO] Fix for SparseCategoricalCrossentropy: cast target to int before one_hot

### PR DESCRIPTION
**Pull request by goyaladitya05**
OpenVINO's `one_hot` op requires integer indices, but target in `sparse_categorical_crossentropy` was passed without casting, causing failures when float labels were provided.

This PR adds casting target to int64 unconditionally at the top of `sparse_categorical_crossentropy`, similar to how other backends do it.

### CODE REVIEW

File: keras/src/backend/openvino/nn.py [hunk 1]
```diff
@@ -843,6 +843,9 @@ def categorical_crossentropy(target, output, from_logits=False, axis=-1):
 
 def sparse_categorical_crossentropy(target, output, from_logits=False, axis=-1):
     target = get_ov_output(target)
+    # one_hot requires integer indices
+    # cast unconditionally, matching JAX/TF/Torch
+    target = ov_opset.convert(target, Type.i64).output(0)
```
[the remark this answers is not in the document]
Comment (goyaladitya05) [reply keras/src/backend/openvino/nn.py:848]: other backends do it similarly.

**Review by hertschuh — APPROVED** [review]
Thanks for the fix!
